wrapper/src/main/java/software/amazon/jdbc/plugin/AwsSecretsManagerConnectionPlugin.java

```diff
@@ -92,14 +97,30 @@ public class AwsSecretsManagerConnectionPlugin extends AbstractConnectionPlugin
   }
 
   public AwsSecretsManagerConnectionPlugin(final PluginService pluginService, final Properties props) {
-
     this(
         pluginService,
         props,
-        (hostSpec, region) -> SecretsManagerClient.builder()
-            .credentialsProvider(AwsCredentialsManager.getProvider(hostSpec, props))
-            .region(region)
-            .build(),
+        (hostSpec, region) -> {
+          final String endpoint = ENDPOINT_PROPERTY.getString(props);
+          if (endpoint != null && !endpoint.isEmpty()) {
+            try {
+              final URI endpointURI = new URI(endpoint);
+              return SecretsManagerClient.builder()
+                  .credentialsProvider(AwsCredentialsManager.getProvider(hostSpec, props))
+                  .endpointOverride(endpointURI)
+                  .region(region)
+                  .build();
+            } catch (URISyntaxException e) {
```

Comment:
is this the only exception we can have from the endpoint override?
What happens if the syntax is ok but the endpoint doesn't lead anywhere?
is there a way to see all the exceptions this call can throw?